app/controllers/timelines_controller.rb

```diff
@@ -360,6 +353,20 @@ def load_timeline_token
       current_ability.options[:timeline_token] = @timeline_token
     end
 
+    def build_url_fragment
+      url_fragment = "noHeader=true&noFooter=true&noSourceLink=false&noVideo=false"
+
+      if @timeline.visibility == 'public' || current_user == @timeline.user
+        url_fragment += "&resource=#{Addressable::URI.escape_component(manifest_timeline_url(@timeline, format: :json), /[:\/?=]/)}"
+        url_fragment += "&callback=#{Addressable::URI.escape_component(manifest_timeline_url(@timeline, format: :json), /[:\/?=]/)}"
+      elsif current_user || @timeline.valid_token?(@timeline.access_token)
```

Comment:
At some point I added the check for `current_user` here but I'm not sure if it is really needed.  Maybe it would be best to turn this line into `else`?  It seems like the access token will be checked in the manifest request (or has already been checked in the `authorize!` call on line 103 and doesn't really need to be checked here.